src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateConstructorFromMembers/GenerateConstructorFromMembersTests.cs

```diff
@@ -392,6 +392,31 @@ public S(int i{|Navigation:)|} : this()
 }");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateConstructorFromMembers)]
+        public async Task TestStructNotInitializingAutoProperty()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System.Collections.Generic;
+
+struct S
+{
+    [|int i { get => f; set => f = value; }|]
+    int j { get; set; }
+}",
+@"using System.Collections.Generic;
+
+struct S
+{
+    public S(int i{|Navigation:)|} : this()
```

Comment:
@jnm2, is this syntax something new in next version of C# language? What is it called? (wana read more about these smilies in c#)

Reply:
This is a special test syntax i created for roslyn when we started.  I wanted a way to create test files that coudl contain special information in-situ.  So we support the following conventions that are understaood by the test harness:

1.  ``$$``.  The position of the caret.
2. ``[|...|]``` an anonymous span.
3. ```{|name:...|}``` a named span.

Test code that can then find these easily.

Reply:
> is this syntax something new in next version of C# language?

Lol, I hope not! 😆